pkg/controller/controlplane/valuesprovider.go

```diff
@@ -207,6 +220,55 @@ func (vp *valuesProvider) decodeControlPlaneConfig(cp *extensionsv1alpha1.Contro
 	if _, _, err := vp.Decoder().Decode(cp.Spec.ProviderConfig.Raw, nil, cpConfig); err != nil {
 		return nil, errors.Wrapf(err, "decoding '%s'", kutil.ObjectName(cp))
 	}
-
 	return cpConfig, nil
 }
+
+// GetShootAdditionalChartValues collects and returns the chart values for all additional charts.
+// It returns them in a map by chart name.
+func (vp *valuesProvider) GetShootAdditionalChartValues(
+	cp *extensionsv1alpha1.ControlPlane,
+	cluster *extensionscontroller.Cluster,
+) (map[string]map[string]interface{}, error) {
+
+	cpConfig, err := vp.decodeControlPlaneConfig(cp)
+	if err != nil {
+		return nil, errors.Wrapf(err, "decoding control plane config")
+	}
+
+	values := map[string]map[string]interface{}{}
+
+	// shoot-rook-ceph
+	var rcValues map[string]interface{}
+
+	if cpConfig.Persistence != nil && *cpConfig.Persistence.Enabled {
+		var (
+			rookCephImage     map[string]interface{}
+			rookCephNamespace string
+		)
+		rookImage, err := imagevector.ImageVector().FindImage(equinixmetal.RookCephImageName)
+		if err == nil {
+			rookCephImage = map[string]interface{}{
+				"repository": rookImage.Repository,
+				"tag":        *rookImage.Tag,
+			}
+		}
+		if cpConfig.Persistence.Namespace != nil {
+			rookCephNamespace = *cpConfig.Persistence.Namespace
+		}
+		if rookCephNamespace == "" {
+			rookCephNamespace = DefaultRookCephNamespace
+		}
```

Comment:
Nit: shorter and a bit more idiomatic
```suggestion
		rookCephNamespace := DefaultRookCephNamespace
		if cpConfig.Persistence.Namespace != nil {
			rookCephNamespace = *cpConfig.Persistence.Namespace
		}
```

Reply:
Well, I guess I could check for both. OK.